boltstub/simple_jolt/transformers.py

```diff
@@ -428,9 +457,12 @@ def _decode_simple(value, decode_cb):
 
     @classmethod
     def _decode_full(cls, value, decode_cb):
-        if not isinstance(value, list) or len(value) != 5:
-            raise JOLTValueError('Expecting list of length 5 after sigil "%s"'
-                                 % cls.sigil)
+        if not isinstance(value, list):
+            raise JOLTValueError('Expecting list after sigil "%s"' % cls.sigil)
+        if len(value) != 5:
+            return JoltRelationTransformer._decode_full_element(value,
+                                                                decode_cb)
+
```

Comment:
This `if` doesn't let clear why the flow is being deviated, Maybe this could be re-write as something like

```
return decode_relationship_with_elementid(value) if if is_relationship_with_element_id(value) else decord_relationship_without_elementid(value)
```

Or something similar, it is also valid for other decode functions.

Reply:
I've added a check if there's an element id at some given index, hopefully that's a bit more clear.

Reply:
It solves part of the issue. 

The old validation and construction logic should be also moved to it owns method. This _decode_full is doing too much thing and it could be a bit confusing. 

All my focus reading these methods are going the construction and validation of the type which occurs after the flow change.

Reply:
I've replaced with  if has field decode v2 else v1.

Reply:
🕺🏼